### ISSUE TITLE
NewlineAtEndOfFile with LF_CR_CRLF does not work properly

### ISSUE THREAD

**Issue by Vampire** [checkstyle/checkstyle#6508]
```
/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="NewlineAtEndOfFile">
        <property name="lineSeparator" value="lf_cr_crlf"/>
    </module>
</module>

/var/tmp $ xxd foo.properties
00000000: 0a                                       .

/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-8.18-all.jar -c config.xml foo.properties
Starting audit...
[ERROR] D:\Sourcecode\other\discord-logger\tmp\foo.properties:1: File does not end with a newline. [NewlineAtEndOfFile]
Audit done.
Checkstyle ends with 1 errors.
```

---------------

With a file that only contains a linefeed character the `NewlineAtEndOfFile` check fails, because `NewlineAtEndOfFile` with `LF_CR_CRLF` requires to have at least two characters in the file with CS 8.18, even when technically only the latter is checked actually.
Expected would be no violation.

https://checkstyle.org/property_types.html#lineSeparator 
`lf_cr_crlf` is "lf, cr or crlf" in other words ANY

**Comment by romani** [comment]
@Vampire , please report issue as required our template.
Thanks in advance.

**Comment by romani** [comment]
@Vampire , is it the same as #4074

**Comment by Vampire** [comment]
No, #4074 is totally different.
#4074 says check is configured to check for LF, file has CRLF which ends in LF and thus is accepted.
This one says check is configured to use LF_CR_CRLF, so any should be accepted, file does only contain one byte which is LF, so should have no validation, but fails as the check for LF_CR_CRLF requires two bytes in the file to match.
I'll update according to the template, one moment.

**Comment by Vampire** [comment]
I updated the inital description with a variation of the report template

**Comment by rnveach** [comment]
I am marking this approved.

**Comment by rnveach** [comment]
Fix was merged
